pyanaconda/modules/payloads/payload/dnf/tree_info.py

```diff
@@ -400,38 +409,33 @@ def enabled(self):
         return self._type in conf.payload.enabled_repositories_from_treeinfo
 
     @property
-    def valid(self):
-        """Is the repository valid?
-
-        We only check whether the repodata directory exists.
-        That doesn't have to mean that the repo is valid.
+    def relative_path(self):
+        """Relative path of the repository.
 
-        :return: True or False
+        :return: a relative path
         """
-        return os.access(os.path.join(self.absolute_path, "repodata"), os.R_OK)
-
-    @property
-    def relative_path(self):
-        """Relative path of the repository."""
         return self._relative_path
 
     @property
-    def absolute_path(self):
-        """Absolute path of the repository."""
-        return self._absolute_path
+    def url(self):
+        """URL of the repository.
+
+        :return: a URL
+        """
+        return self._url
 
     @staticmethod
-    def _get_absolute_path(root_path, relative_path):
-        """Get the absolute path of the repository."""
+    def _get_url(root_url, relative_path):
+        """Get the URL of the repository."""
         if relative_path == ".":
-            return root_path
+            return root_url
+
+        # Get the protocol.
+        protocol, root_path = split_protocol(root_url)
 
         # Create the absolute path.
-        full_path = os.path.join(root_path, relative_path)
-        protocol, url = split_protocol(full_path)
+        absolute_path = os.path.join(root_path, relative_path)
```

Comment:
Did you want to use `pyanaconda.core.path.join_paths` here?

Reply:
Fixed in https://github.com/rhinstaller/anaconda/pull/4047.